src/coreComponents/common/logger/ErrorHandling.cpp

```diff
@@ -0,0 +1,252 @@
+  }
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::addToMsg( std::exception const & e, bool toEnd )
+{
+  if( toEnd )
+  {
+    m_msg = m_msg + e.what();
+  }
+  else
+  {
+    m_msg = e.what() + m_msg;
+  }
+  return *this;
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::addToMsg( std::string_view errorMsg, bool toEnd )
+{
+  if( toEnd )
+  {
+    m_msg = m_msg + std::string( errorMsg );
+  }
+  else
+  {
+    m_msg = std::string( errorMsg ) + m_msg;
+  }
+  return *this;
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::setCodeLocation( std::string_view msgFile, integer msgLine )
+{
+  m_file = msgFile;
+  m_line = msgLine;
+  return *this;
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::setType( ErrorLogger::MsgType msgType )
+{
+  m_type = msgType;
+  return *this;
+}
+
+void ErrorLogger::ErrorMsg::addContextInfoImpl( ErrorLogger::ErrorContext && ctxInfo )
+{
+  m_contextsInfo.emplace_back( std::move( ctxInfo ) );
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::setRank( int rank )
+{
+  m_ranksInfo.push_back( rank );
+  return *this;
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::addCallStackInfo( std::string_view ossStackTrace )
+{
+  std::string str = std::string( ossStackTrace );
+  std::istringstream iss( str );
+  std::string stackLine;
+  std::size_t index;
+
+  std::regex pattern( R"(Frame \d+: \S+)" );
+
+  while( std::getline( iss, stackLine ) )
+  {
+    if( std::regex_search( stackLine, pattern ))
+    {
+      m_isValidStackTrace = true;
+      index = stackLine.find( ':' );
+      m_sourceCallStack.push_back( stackLine.substr( index + 1 ) );
+    }
+  }
+
+  if( !m_isValidStackTrace )
+  {
+    m_sourceCallStack.push_back( std::string( g_callStackMessage ) );
+  }
+
+  return *this;
+}
+
+std::string ErrorLogger::toString( ErrorLogger::MsgType type )
+{
+  switch( type )
+  {
+    case ErrorLogger::MsgType::Error: return "Error";
+    case ErrorLogger::MsgType::Warning: return "Warning";
+    case ErrorLogger::MsgType::Exception: return "Exception";
+    default: return "Unknown";
+  }
+}
+
+void ErrorLogger::streamMultilineYamlAttribute( std::string_view msg, std::ofstream & yamlFile,
+                                                std::string_view indent )
+{
+  std::size_t i = 0;
+  // Loop that runs through the string_view named msg
+  while( i < msg.size() )
+  {
+    // Index of the next line break
+    std::size_t index = msg.find( "\n", i );
+    // If there is no line break, the entire string is taken
+    if( index == std::string_view::npos )
+    {
+      index = msg.size();
+    }
+    // Writes the current line to the YAML file with the desired indentation
+    std::string_view msgLine = msg.substr( i, index - i );
+    yamlFile << indent << msgLine << "\n";
+    // Move to the next line
+    i = index + 1;
+  }
+}
+
+void ErrorLogger::flushErrorMsg( ErrorLogger::ErrorMsg & errorMsg )
+{
+  std::ofstream yamlFile( std::string( m_filename ), std::ios::app );
```

Comment:
This will create the file. I think we should do this only if after checking `isOutputFileEnabled()`